### DISCUSSION TITLE
Automatization Solidity smart contracts analysis

### DISCUSSION THREAD

**Discussion by mstad** [trailofbits/manticore#1730] [General]
Good morning, I am Mirko, a PhD candidate and I would automatize the analysis of some smart-contracts mutations, I try to individuate automatically contracts (in a *.sol file) for manticore analysis, and then running 'manticore file.sol --contract ContractName'. My problem is that in some cases manticore returns an error in which it informs me that it analyzes only contracts that use CREATE. How can I overcome this problem? Can I recognize contracts (analyzing the code of the contract) that Manticore can not analyze (and eventually in which way)... In case of a negative reply, I have to select contracts by hand, but it takes so much time. Thanks in advance.

**Comment by feliam** [comment]
Hello @mstad. Tank you for trying ManticoreEVM! 
I am not sure I understand the error manticore is throwing at you. 
Random datapoint. Internally Manticore does not _need_ the source code. It emulates the lowlevel ethereum architechture fully (to call it some way).  Though the commandline manticore interface does actually need you to provide the sourcecode in certain form and to select a target contract. It makes some assumtions to make the most common case easier. 

So I guess we'll need to see exactly what are you seeing to properly diagnose this. So please provide more detail about what is happening. What is the contract? What is the exact commandline? Stuff like that so we can try to reproduce it on our side.

Alternatively you can reach us in this instant messaging thing: https://empireslacking.herokuapp.com/ (channel #manticore) If you want faster responses!
